tests/spoke/spoke_suite_test.go

```diff
@@ -144,6 +146,19 @@ var _ = BeforeSuite(func() {
 	}).SetupWithManager(k8sManager)
 	Expect(err).ToNot(HaveOccurred())
 
+	err = (&namespace.Reconciler{
+		Client: k8sManager.GetClient(),
+		Scheme: k8sManager.GetScheme(),
+		Log:    ctrl.Log.WithName("NamespaceTest"),
+		EventRecorder: &events.EventRecorder{
```

Comment:
Why not use an actual event recorder like 
`namespaceEventRecorder := events.NewEventRecorder(mgr.GetEventRecorderFor("namespace-controller"))`
?

Reply:
In testing everywhere fake event recorder is used

Reply:
@rahulsawra98 [here](https://github.com/kubeslice/worker-operator/blob/5ff08e964cea463f3f856942c57773c0962c7d50/tests/hub/hub_suite_test.go#L114) we are using EventRecorder from pkg/events, we can use it for tests/spoke as well right ?

Reply:
but could you please tell me what is the problem with using a fake one? Because we are creating any events as of now

Reply:
yaa it makes sense to use it in case you are testing whether the actual events got created or not!
In case you dont want to test, fakeRecorder is also fine!
Anything works here

Reply:
@YachikaRalhan in one of my earlier test cases in tests/hub, I noticed crashes with fake event recorder so just making sure about this with @rahulsawra98